packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -277,6 +277,11 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
     rowClassNames.push(rootProps.getRowClassName(rowParams));
   }
 
+  /* Start of rendering */
+  if (!rowNode) {
+    return null;
+  }
+
```

Comment:
Is moving this block required?

I prefer to keep functional components organized as if it were a class component:

```
class GridRow {
  constructor() { ... }
  onClick() { ... }
  getCell() { ... }
  render() {}
}
```

```
function GridRow {
  /* initialization code */
  const onClick = () => { ... }
  const getCell = () => { ... }
  /* render code */
}
```

In particular for large components, it helps readability a lot to keep all render code grouped rather than spread in-between "class methods".

Reply:
I like grouping logic, but here the balance is between a pretty large dead code path (defining an anyonymous function that should never be called, unless rowNode exists) vs. grouping logic. Not that clear what's more readable.

After this change, if you were to accidentally remove the rowNode check on component-level, the types would break (good imho), which is better than adding an extra conditional in the getCell that would open up the potential for using the compnent in currently unintended/undesigned ways.

Probably the right question here to ask is why the component needs to (try to) mount without a rowNode to begin with.

Reply:
> Probably the right question here to ask is why the component needs to (try to) mount without a rowNode to begin with.

Absolutely, the answer is often that the codebase went through many architectural & maintainer changes, and we haven't got around to refactoring everything.